add new events and event properties (#4189)
segmentio/segment-docs: src/connections/sources/catalog/cloud-apps/braze/index.md

| Email Bounced                          | Braze attempted to send an email, but the User’s receiving mail server did not accept it.                                                                 |
| Email Soft Bounced                     | Braze attempted to send an email, but the User’s receiving mail server temporarily bounced it. (Reasons may include: inbox full, server down, and so on.) |
| Email Marked As Spam                   | User marked an email as spam.                                                                                                                             |
| Email Unsubscribed                     | User clicked the unsubscribe link in an email.                                                                                                            |
| Global Subscription Group State Change | User’s subscription global group state changed to `Subscribed` or `Unsubscribed`.                                                                         |
| Subscription Group State Changed       | User’s subscription group state changed to `Subscribed` or `Unsubscribed`.                                                                                |

> | Push Notification Sent                 | A push notification was successfully sent                                                                                                                 |

Reviewer: ```suggestion
| Push Notification Sent                 | A push notification was successfully sent.                                                                                                                 |
```

| iOS Foreground Push Opened             | User received a push notification on iOS while the app was open.                                                                                          |
| In-App Message Viewed                  | User viewed an in-app message.                                                                                                                            |
| In-App Message Clicked                 | User tapped or clicked a button in an in-app message.                                                                                                     |
| News Feed Viewed                       | User viewed the native Braze News Feed.                                                                                                                   |
| News Feed Card Viewed                  | User viewed a Card within the native Braze News Feed.                                                                                                     |
| News Feed Card Clicked                 | User tapped or clicked on a Card within the native Braze News Feed.                                                                                       |

> | Webhook Sent                           | A webhook message was sent                                                                                                                                |

Reviewer: ```suggestion
| Webhook Sent                           | A webhook message was sent.                                                                                                                                |
```

| Content Card Sent                      | A Content Card was sent to a user's device.                                                                                                               |
| Content Card Viewed                    | User viewed a Content Card.                                                                                                                               |
| Content Card Clicked                   | User clicked a Content Card.                                                                                                                              |
| Content Card Dismissed                 | User dismissed a Content Card.                                                                                                                            |
| SMS Sent                               | An SMS was sent.                                                                                                                                          |
| SMS Sent to Carrier                    | An SMS was sent to the Carrier for delivery.                                                                                                              |

> | SMS Short Link Clicks                  | Short link in an SMS message is clicked                                                                                                                   |

Reviewer: ```suggestion
| SMS Short Link Clicks                  | User clicked a short link in an SMS message.                                                                                                                |
```

| `campaign_id`          | String | The API Identifier of the campaign associated with the event, if applicable.                                                                                                                                                                   |
| `campaign_name`        | String | The name of the campaign associated with the event, if applicable.                                                                                                                                                                             |
| `message_variation_id` | String | The API Identifier of the Message Variation for the campaign associated with the event, if applicable.                                                                                                                                         |

> | `message_variation_name`      | String | To further usability, in all events where we provide Message Variation ID, we'll now also provide the Message Variation Name.                                                                                                                                                                |

Reviewer: ```suggestion
| `message_variation_name`      | String | To further usability, in all events where Segment provides Message Variation ID, Segment now also provides the Message Variation Name.                                                                                                                                                                |
```

| `link_url`              | String | For Email Clicked events, the URL of the link that the user clicked on.                                                |
| `button_id`             | String | For In-App Message Clicked events, the index of the button the user clicked on.                                        |
| `card_id`               | String | For News Feed Card and Content Card events, the API Identifier of the Card.                                            |

> | `state_change_source`   | String | Source of the state change (i.e., REST, SDK , Dashboard, Preference Center, etc. )                                     |

Reviewer: ```suggestion
| `state_change_source`   | String | Source of the state change (for example, REST, SDK, Dashboard, Preference Center, and more.)                                     |
```
